homeassistant/components/notify/legacy.py

```diff
@@ -32,7 +33,9 @@
 NOTIFY_DISCOVERY_DISPATCHER = "notify_discovery_dispatcher"
 
 
-async def async_setup_legacy(hass: HomeAssistant, config: ConfigType) -> None:
+async def async_setup_legacy(
```

Comment:
This doesn't look like it needs to be a coroutine function anymore. It could be a callback and then we know it will not yield to the loop before it returns.

Reply:
Good idea, cd4370a4c2b50b1c0eec3fa4f1f776eb979f2d72